orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -209,10 +206,51 @@ class RunTaskHandler(
   private fun Task.backoffPeriod(taskModel: com.netflix.spinnaker.orca.pipeline.model.Task, stage: Stage): TemporalAmount =
     when (this) {
       is RetryableTask -> Duration.ofMillis(
-        Math.min(getDynamicBackoffPeriod(stage, Duration.ofMillis(System.currentTimeMillis() - (taskModel.startTime
-          ?: 0))), maxBackoff())
+        retryableBackOffPeriod(taskModel, stage).coerceAtMost(taskExecutionInterceptors.maxBackoff())
       )
-      else -> Duration.ofSeconds(1)
+      else -> Duration.ofMillis(1000)
+    }
+
+  /**
+   * The max back off value always wins.  For example, given the following dynamic configs:
+   * tasks.global.backOffPeriod = 5000
+   * tasks.aws.backOffPeriod = 80000
+   * tasks.aws.someAccount.backoffPeriod = 60000
+   * tasks.aws.backoffPeriod will be used (given the criteria matches and unless the default dynamicBackOffPeriod is greater).
+   */
+  private fun RetryableTask.retryableBackOffPeriod(
+    taskModel: com.netflix.spinnaker.orca.pipeline.model.Task,
+    stage: Stage
+  ): Long {
+    val dynamicBackOffPeriod = getDynamicBackoffPeriod(stage, Duration.ofMillis(System.currentTimeMillis() - (taskModel.startTime ?: 0)))
+    val backOffs: MutableList<Long> = mutableListOf(
+      dynamicBackOffPeriod,
+      dynamicConfigService.getConfig(Long::class.java, "tasks.global.backOffPeriod", dynamicBackOffPeriod)
+    )
+
+    if (this is CloudProviderAware) {
+      if (hasCloudProvider(stage)) {
```

Comment:
`if (this is CloudProviderAware && hasCloudProvider(stage)` rather than two `if`s.